File: src/main/java/com/shelfify/shelfifyapi/ean/EanMappingRepository.java
package com.shelfify.shelfifyapi.ean;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.math.BigInteger;
import java.util.List;
import java.util.Optional;

@Repository
public interface EanMappingRepository extends JpaRepository<EanMapping, Long> {
    Optional<EanMapping> findByEanAndDatagroupIsNull(String ean);

    Optional<EanMapping> findByEanAndDatagroup(String ean, String datagroup);

    Optional<EanMapping> findByProductNameAndDatagroup(String name, String datagroup);

    Optional<EanMapping> findByProductNameAndDatagroupIsNull(String name);
}

File: src/main/kotlin/de/schnitzel/shelfifyapi/repositories/EanMappingRepository.kt
package de.schnitzel.shelfifyapi.repositories

import de.schnitzel.shelfifyapi.entities.EanMappingEntity
import org.springframework.data.jpa.repository.JpaRepository

interface EanMappingRepository : JpaRepository<EanMappingEntity, Int> {
    fun findByEan(ean: String): EanMappingEntity?
}